packages/zoe/src/zoeService/instanceAdminStorage.js

```diff
@@ -1,49 +1,284 @@
 import { E } from '@endo/eventual-send';
-import { makeWeakStore } from '@agoric/store';
+import {
+  canBeDurable,
+  makeScalarBigSetStore,
+  provideDurableWeakMapStore,
+  vivifyKindMulti,
+  vivifyFarClassKit,
+  M,
+  provide,
+} from '@agoric/vat-data';
+import { makeZoeSeatAdminFactory } from './zoeSeat.js';
+import { defineDurableHandle } from '../makeHandle.js';
+import { InstanceAdminI, InstanceHandleShape } from '../typeGuards.js';
 
-/**
- *
- */
-export const makeInstanceAdminStorage = () => {
-  /** @type {WeakStore<Instance,InstanceAdmin>} */
-  const instanceToInstanceAdmin = makeWeakStore('instance');
+const { quote: q, details: X } = assert;
 
-  /** @type {import('./utils.js').GetPublicFacet} */
-  const getPublicFacet = async instanceP =>
-    E.when(instanceP, instance =>
-      instanceToInstanceAdmin.get(instance).getPublicFacet(),
-    );
+const InstanceAdminStorageShape = M.interface('InstanceAdminStorage', {
+  getPublicFacet: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
+  getBrands: M.call(InstanceHandleShape).returns(M.promise()),
+  getIssuers: M.call(InstanceHandleShape).returns(M.promise()),
+  getTerms: M.call(InstanceHandleShape).returns(M.promise()),
+  getOfferFilter: M.call(M.eref(InstanceHandleShape)).returns(M.promise()),
+  setOfferFilter: M.call(M.eref(InstanceHandleShape)).returns(),
+  getInstallationForInstance: M.call(InstanceHandleShape).returns(M.promise()),
+  getInstanceAdmin: M.call(InstanceHandleShape).returns(InstanceAdminI),
```

Comment:
You cannot use an interface guard in this position. You must use a pattern

```suggestion
  getInstanceAdmin: M.call(InstanceHandleShape).returns(InstanceAdminShape),
```